test/content-negotiation-test.js

```diff
@@ -0,0 +1,61 @@
+var test = require('tape');
+var Hapi = require('hapi');
+var server = new Hapi.Server();
+var routes = require('../routes.js');
+
+server.connection({
+  port: process.env.TESTPORT || '8080'
+});
+
+server.route(routes);
+
+server.register(require('hapi-negotiator'), function (err) {
+  if (err) {
+    console.error('Failed to load plugin:', err);
+  }
+});
+
+test('Content Negotiation', function (t) {
+  var htmlRequest = {
+    method: 'GET',
+    url: '/',
+    headers: {'Accept': 'text/html'}
+  };
+
+  var jsonRequest = {
+    method: 'GET',
+    url: '/',
+    headers: {'Accept': 'application/vnd.api+json'}
+  };
+
+  var badJSONRequest = {
+    method: 'GET',
+    url: '/',
+    headers: {'Accept': 'application/vnd.api+json; charset=utf-8'}
+  };
+
+  var acceptableJSONRequest = {
+    method: 'GET',
+    url: '/',
+    headers: {'Accept': 'application/vnd.api+json; charset=utf-8, application/vnd.api+json'}
+  };
+
+  server.inject(htmlRequest, function (res) {
+    t.equal(res.payload, '<h1>HTML Response</h1>', 'HTML request correct');
```

Comment:
It won't be possible to test the full response payload payload like this in the future. I suggest we look for HTML we know will be there, like `<!doctype html>` for example.